include/oneapi/tbb/detail/_machine.h

```diff
@@ -75,12 +77,12 @@ using std::this_thread::yield;
 // atomic_fence implementation
 //--------------------------------------------------------------------------------------------------
 
-#if (_MSC_VER)
+#if (_MSC_VER) && _M_X64
 #pragma intrinsic(_mm_mfence)
 #endif
 
 static inline void atomic_fence(std::memory_order order) {
-#if (_WIN32)
+#if (_WIN32) && _M_X64
```

Comment:
```suggestion
#if _MSC_VER && (__TBB_x86_64 || __TBB_x86_32)
```
Fix comment on line 94 as well